packages/catalog-process/src/services/catalogService.ts

```diff
@@ -843,6 +844,79 @@ export function catalogServiceBuilder(
       return updatedEService;
     },
 
+    async updateEServiceInstance(
+      eserviceId: EServiceId,
+      eserviceSeed: catalogApi.UpdateEServiceInstanceSeed,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<EService> {
+      logger.info(`Updating EService ${eserviceId} Instance`);
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        authData,
+        readModelService
+      );
+
+      if (eservice.data.templateRef === undefined) {
+        throw eServiceNotAnInstance(eserviceId);
+      }
```

Comment:
You could create a dedicated validator for this `assertEServiceIsTemplateInstance`